### ISSUE TITLE
Bug: Market order submission not returning 'Insufficient funds' error

### ISSUE THREAD

**Issue by fujiapple852** [coinbase/coinbase-pro-trading-toolkit#199]
Hi,

As with my [previous bug report](https://github.com/coinbase/gdax-tt/issues/195) this appear to be an exchange side bug rather than a bug with this client library (which I do not use) and so if there is a more appropriate place to report this please let me know.

### The issue
Submitting market orders whilst not having sufficient funds does not always result in an 'Insufficient funds' error response.  

The precise circumstances under which this occurs is not completely clear, from experimentation it appears to occur for 'buy' market orders but not for 'sell' market orders.

I also observe that when this problem occurs we are sent a 'received' message for the order however the 'size' field is 0.

In summary:

Sell BTC-USD (with zero BTC in account) -> 'Insufficient funds' error returned from POST call
Buy BTC-USD (with zero USD in account) -> no error returned from POST call but 'received' shows size as 0

See also [this bug report](https://github.com/danpaquin/gdax-python/issues/280) for another example of this issue occurring when using market orders with 'size' but not occurring when using 'funds'.

Note this issue was observed on the 'sandbox' environment, I have not attempted to reproduce it on the live trading system.

If needed I can produce a minimal reproducible example to demonstrate the bug. 

Thanks,

Fuji Apple

**Comment by sfkiwi** [comment]
Out of curiosity how does your order look like? Are you using size or funds field for you market buy order. I ask because limit buy, sell and market sell orders all use size whereas market buy orders use funds field in the order message. 

I just tested on the live exchange and placing a market order with zero USD fails to place when specifying the order as `funds: $100` but succeeds as a $0 market order when specifying the order as `size: 0.002`. I believe this is because specifically for market buy orders gdax is looking for the funds field and if it doesn't see it is setting to max available balance. I tested this by placing a market buy order for `size: 0.1` (around $700) but with only $10 in my USD balance. The order succeeds using the total available balance so basically they are automatically fitting it to your available balance and a $0 balance is included as an acceptable value.

**Comment by fujiapple852** [comment]
@sfkiwi - thank you for your reply.  Your testing results are inline with what @mfxuus reported in the issue linked above (i.e. works for `funds` but not always for `size`).  

In my case I am only using `size` for both buy and sell orders which, as i interpret [the documentation](https://docs.gdax.com/#place-a-new-order), is valid:

> *market order parameters*
> ````
> size | [optional]* Desired amount in BTC
> funds | [optional]* Desired amount of quote currency to use
> ````
> * One of size or funds is required.

I could of course calculate the corresponding `funds` for a given buy order `size` to workaround the bug (as it happens I have another workaround in place) but I think it would be best for GDax to fix the issue.  

As an aside, I notice in the GDax web UI that for buy orders you are _only_ able to specify funds (i.e. USD amount to sell) which I find very non-intuitive - I guess it is considered more user friendly as it is similar to how retail _Bureau de Change_ works which people may be more comfortable with rather than FX markets conventions of buying/selling the FX ticker.

Thanks,

Fuji Apple

**Comment by drewrothstein** [comment]
Hi, we are closing out PRs + Issues as this project is being archived.
